### PR TITLE
Add Python 3.10 tests to GitHub Actions

### LINKED ISSUE #1250: Add Python 3.10 tests to GitHub Actions
After Python 3.10 is released in ∼October, we'll need to add it to our test suite.  Python 3.10 has significant improvements to common error messages, so it would be worth pointing contributors to the Python 3.10 test results as the first place to go to for debugging test failures (which I'm mentioning in #1163 too).

Adding this feature now because I'm excited to use some of the new features in 3.10...in approximately 43 months.

### PR DESCRIPTION
**Pull request by namurphy**
This pull request is to add Python 3.10 to our test suite.  Python 3.10 is due to be released approximately today 🥳 🎈 📈 but this PR will probably not work yet so we should wait to merge it.

- [x] I have added a changelog entry for this pull request.

- [x] If adding new functionality, I have added tests and
      docstrings.

- [x] I have fixed any newly failing tests.

Closes #1250.

### CODE REVIEW

**Comment by namurphy** [comment]
pre-commit.ci autofix

**Comment by StanczakDominik** [comment]
pre-commit.ci autofix

I think the arrow part messed it up and it might need to be on a single line!

**Comment by StanczakDominik** [comment]
Now stalled on numpy not yet providing 3.10 wheels.

**Comment by rocco8773** [comment]
Note that these test failures are primarily due to `numba`'s current incompatibility with python 3.10 (https://github.com/numba/numba/issues/7562).

**Comment by StanczakDominik** [comment]
Yup, still stalled on numba.

**Comment by StanczakDominik** [comment]
For the record, numba can now be installed on 3.10 using
```python
pip install --pre llvmlite
pip install "git+git://github.com/numba/numba.git"
```

Let's wait for them to make the full release, though.

**Comment by namurphy** [comment]
Numba 0.55.0 was just released a few days ago, so if all of the tests pass then we could merge this pretty soon.  Here's hoping!

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -67,6 +67,11 @@ jobs:
           python: 3.8
           toxenv: py38-all-cov
 
+        - name: Python 3.10
+          os: ubuntu-latest
+          python: "3.10"
+          toxenv: py310
+
```
Comment (namurphy) [inline .github/workflows/testing.yml:74]: - [ ] Move this to end of the list

**Review by rocco8773 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**